### PR TITLE
allow configration of Mattermost Username

### LINKED ISSUE #245: [BUG] MatterMost getUser call is causing index out of range exception
As per the documentation, MatterMost integration is not working. After providing the details in values.yml, deploying the helm chart is causing the pods in `CrashLoopBackOff` status.

1. Please proceed with the MatterMost integration as described here: https://www.botkube.io/installation/mattermost/
2. Set the logLevel to `INFO`
3. Deploy the helm chart in kubernetes cluster v 1.15.4
4. Check the status of the pod and grep the console log.

**Versions**
MatterMost: v5.9.8
Kubernetes: v1.15.4
Botkube: v0.9.1

**Expected behavior**
The pod should be in `Running` status.

**Screenshots**
MatterMost

**Console Log**
```
INFO[2020-02-11T22:15:20Z] BotKube connected to Mattermost!             
INFO[2020-02-11T22:15:20Z] Registering watcher on configfile /config/config.yaml 
panic: runtime error: index out of range

goroutine 11 [running]:
github.com/infracloudio/botkube/pkg/bot.MMBot.getUser(0xc00036a5c0, 0x1a, 0xc00033d320, 0x4, 0xc00033d2d0, 0x6, 0xc00033d778, 0x3, 0x1, 0xc00036a620, ...)
	/app/pkg/bot/mattermost.go:193 +0x191
github.com/infracloudio/botkube/pkg/bot.MMBot.listen(0xc00036a5c0, 0x1a, 0xc00033d320, 0x4, 0xc00033d2d0, 0x6, 0xc00033d778, 0x3, 0x1, 0xc00036a620, ...)
	/app/pkg/bot/mattermost.go:226 +0x252
github.com/infracloudio/botkube/pkg/bot.(*MMBot).Start.func1(0xc000241c00)
	/app/pkg/bot/mattermost.go:107 +0xf0
created by github.com/infracloudio/botkube/pkg/bot.(*MMBot).Start
	/app/pkg/bot/mattermost.go:95 +0x429
```

### PR DESCRIPTION
**Pull request by goir**
##### ISSUE TYPE
 - Bug fix Pull Request

##### SUMMARY
Allow Configuration of Mattermost user and fixes BUG with getUser() if its not the "BotKube" default user.

Fixes #245

### CODE REVIEW

File: helm/botkube/values.yaml [hunk 1]
```diff
@@ -278,6 +278,7 @@ communications:
   # Settings for Mattermost
   mattermost:
     enabled: false
+    botname: 'BotKube'                          # User in mattermost which belongs the Personal Access token to
```
Comment (PrasadG193) [inline helm/botkube/values.yaml:281]:
```suggestion
    botName: 'BotKube'                          # User in mattermost which belongs the Personal Access token to
```

**Review by PrasadG193 — COMMENTED** [review]
Thanks for the PR @goir. I just have a minor comment, the rest looks good.

**Comment by goir** [comment]
Hi @PrasadG193 sorry for the delay. I've changed the name as you suggested. If there is anything else, just let me know.
Happy this gets merged :)

File: pkg/config/config.go [hunk 2]
```diff
@@ -176,6 +176,7 @@ type Index struct {
 // Mattermost configuration to authentication and send notifications
 type Mattermost struct {
 	Enabled   bool
+	BotName   string
```
Comment (PrasadG193) [inline pkg/config/config.go:179]: @goir I believe, we would need to add yaml tag here `yaml: "botName"`. Could you please verify if the botName from config is getting mapped in the struct correctly?

Comment (goir) [reply pkg/config/config.go]: Hi @PrasadG193 you were right. had to add it, sorry i'm still learning go. But its working now. thank you

**Review by PrasadG193 — APPROVED** [review]

**Pull request merged**